docs/data-sources/vlan.md

```diff
@@ -0,0 +1,60 @@
+---
+page_title: "Equinix Metal: metal_vlan"
+subcategory: ""
+description: |-
+  Provides an Equinix Metal Virtual Network datasource. This can be used to read vlans.
+---
+
+# metal_vlan
+
+Provides an Equinix Metal Virtual Network datasource. Users can look up VLANs based on either VLAN UUID, or project UUID and vxlan number.
+
+## Example Usage
+
+Fetch a vlan by ID:
+
+```hcl
+resource "metal_vlan" "foovlan" {
+        project_id = local.project_id
+        metro = "sv"
+        vxlan = 5
+}
+
+data "metal_vlan" "dsvlan" {
+        vlan_id = metal_vlan.foovlan.id
+}
+```
+
+Fetch a vlan by project ID, vxlan and metro
+
+```hcl
+resource "metal_vlan" "foovlan" {
+        project_id = local.project_id
+        metro = "sv"
+        vxlan = 5
+}
+
+data "metal_vlan" "dsvlan" {
+        project_id = local.project_id
+        vxlan      = 5
+        metro      = "sv"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `vlan_id` - Metal UUID of the VLAN resource to look up
+* `project_id` - UUID of parent project of the VLAN. Use together with the vxlan number and metro or facility`
```

Comment:
trailing backtick here on project_id.

I think we generally say optional or required (or both?)  when listing the arguments

Reply:
Everything is optional, I was thinking project_id was required, but not in the presence of vlan_id. I suppose this is the usual format then. Perhaps we could add that Either vlan_id or project_id is required?